### DISCUSSION TITLE
Allow errorLevel="<int>" on individual files and folders

### DISCUSSION THREAD

**Discussion by olleharstedt** [vimeo/psalm#7510] [Ideas]
This could be very handy for our legacy code-base where we apply Psalm step by step. Thoughts?

Example:

```xml
    <projectFiles>
        <directory errorLevel="8" name="controllers" />
        <!-- This applies errorLevel as defined for the whole doc -->
        <directory name="models" />
    </projectFiles>
```

There's a six year old issue here: https://github.com/vimeo/psalm/issues/21

**Comment by orklah** [comment]
That could be interesting and I don't think it would be very hard to do.

If you're up to it, I'd suggest starting in Config::getReportingLevelForFile

**Comment by olleharstedt** [comment]
I might :)
